### PR TITLE
Support new registration properties.

### PR DESCRIPTION
**Pull request by rayhayes**

### CODE REVIEW

File: DSCResources/MSFT_xDSCWebService/MSFT_xDSCWebService.Schema.mof [hunk 1]
```diff
@@ -1,15 +1,16 @@
...
+  [write] string PhysicalPath;
+  [write,ValueMap{"Present", "Absent"},Values{"Present", "Absent"}] string Ensure;
+  [write,ValueMap{"Started","Stopped"},Values{"Started", "Stopped"}] string State;
+  [write] string ModulePath;
+  [write] string ConfigurationPath;
+  [write] boolean IsComplianceServer;
+  [read] string DSCServerUrl;  
+  [write] string RegistrationKeyPath;
+  [write] boolean AcceptSelfSignedCertificates;
+};
```
Comment (vors) [inline DSCResources/MSFT_xDSCWebService/MSFT_xDSCWebService.Schema.mof]: please add new line at the end.

File: DSCResources/MSFT_xDSCWebService/PSWSIISEndpoint.psm1 [hunk 2]
```diff
@@ -1,664 +1,663 @@
...
-            $xml.configuration.AppendChild($runtimeSetting)
-
-            $xml.Save($webconfig) 
-        }
-    }
-}
-
-Export-ModuleMember -function New-PSWSEndpoint, Set-AppSettingsInWebconfig, Set-BindingRedirectSettingInWebConfig
-
+﻿# This module file contains a utility to perform PSWS IIS Endpoint setup
```
Comment (vors) [inline DSCResources/MSFT_xDSCWebService/PSWSIISEndpoint.psm1]: I suspect this change caused by line endings change. We should consider adding test for that to common tests.

**Comment by vors** [comment]
Awesome, can you add some tests?

File: DSCResources/MSFT_xDSCWebService/MSFT_xDSCWebService.Schema.mof (hunk 1 above)
Comment (rayhayes) [inline DSCResources/MSFT_xDSCWebService/MSFT_xDSCWebService.Schema.mof]: Done.

**Comment by rayhayes** [comment]
Yes. I need to get up to speed on how to write tests that run in git. Is there a document?

**Comment by vors** [comment]
Sort of :) https://github.com/PowerShell/DscResources/blob/master/CONTRIBUTING.md#improving-test-coverage-for-existing-resources we are discussing conventions that would make sense here: https://github.com/PowerShell/DscResources/issues/9

**Comment by rayhayes** [comment]
Thanks. I'll give it a look and see what I can come up with.

File: DSCResources/MSFT_xDSCWebService/MSFT_xDSCWebService.psm1 [hunk 3]
```diff
@@ -196,7 +221,27 @@ function Set-TargetResource
...
+        PSWSIISEndpoint\Set-AppSettingsInWebconfig -path $PhysicalPath -key "ModulePath" -value $ModulePath
+
+        $null = New-Item -path "$RegistrationKeyPath" -itemType "directory" -Force
+
+        PSWSIISEndpoint\Set-AppSettingsInWebconfig -path $PhysicalPath -key "RegistrationKeyPath" -value $RegistrationKeyPath
+
+        if($AcceptSelfSignedCertificates)
+        {
+            Copy-Item "$pathPullServer\IISSelfSignedCertModule.dll" %windir%\System32\inetsrv -Force
+            Copy-Item "%windir%\SysWOW64\WindowsPowerShell\v1.0\Modules\PSDesiredStateConfiguration\PullServer\IISSelfSignedCertModule.dll" %windir%\SysWOW64\inetsrv -Force
```
Comment (TravisEz13) [inline DSCResources/MSFT_xDSCWebService/MSFT_xDSCWebService.psm1]: powershell will not evaluate %windir%

File: DSCResources/MSFT_xDSCWebService/PSWSIISEndpoint.psm1 [hunk 4]
```diff
@@ -1,664 +1,663 @@
...
+
+            # The <runtime> section goes inside <Configuration> section
+            $xml.configuration.AppendChild($runtimeSetting)
+
+            $xml.Save($webconfig) 
+        }
+    }
+}
+
+Export-ModuleMember -function New-PSWSEndpoint, Set-AppSettingsInWebconfig, Set-BindingRedirectSettingInWebConfig
```
Comment (TravisEz13) [inline DSCResources/MSFT_xDSCWebService/PSWSIISEndpoint.psm1]: should end the file with a new line, see [style guidelines](https://github.com/PowerShell/DscResources/blob/master/StyleGuidelines.md#general-rules)

**Pull request merged**